### PR TITLE
release-notes.txt: add 2024.10 release notes

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This adds the release notes for the 2024.10 release.

### Testing procedure

Please read and see if something is missing.

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,510 @@
...
+  if… (#20834)
+* gnrc_ipv6_nib: Force unspecified next hop addresses (#20371)
+* gnrc_sixlowpan_frag: enable `gnrc_netif_pktq` if `netdev_new_api` is
+  used (#20879)
+* net/dhcpv6: Improve option handling in dhcpv6 advertise (#20801)
+
+Packages
+--------
+
+* makefiles/docker.inc.mk: bump riotbuild docker image to last
```
Comment (maribu) [inline release-notes.txt:87]: This looks miscategirized to me

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,510 @@
...
+------
+
++ boards/adafruit-grand-central-m4-express: provide arduino features (#20904)
++ boards/adafruit-metro-m4-express: initial port (#20912)
++ boards/nucleo-f439zi: add ADC support (#20814)
++ boards/nucleo-g474re: add MCU table (#20851)
++ boards/nucleo64: add pinout source (#20931)
++ boards/stm32l476g-disco: add ADC support (#20793)
++ boards/stm32l476g-disco: add PWM support (#20807)
++ examples: Add asynchronous Rust example (#20830)
```
Comment (Teufelchen1) [inline release-notes.txt:107]: Thats not board specific?

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,510 @@
...
+
+* cpu/stm32: implement `periph_gpio_ll_switch_dir` (#20805)
+* drivers/periph_gpio: let gpio_read() return bool (#20936)
+* drivers/periph_gpio_ll: change API to access GPIO ports (#20639)
+
+Uncategorized
+-------------
++ .clang-format: Add default clang-format configuration (#20865)
++ CODING_CONVENTIONS.md: Add preprocessor directive formatting (#20866)
+* riot.doxyfile: Update and remove deprecated options (#20897)
```
Comment (miri64) [inline release-notes.txt:185]: This belongs to `Documentation`.

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,510 @@
...
+* core/mutex: use thread_yield_higher() in mutex_unlock() (#20890)
+* core/panic: don't use LOG_ functions in panic handler (#20790)
+* cpu/sam0_common/periph_gpio_ll: fix gpio_get_port() and
+  gpio_ll_query_conf() [backport 2024.10] (#21008)
+* dns_msg: skip RDLENGTH_LENGTH field when skipping record (#20857)
+* drivers/at86rf215: return ENETDOWN when interface is down [backport
+  2024.10] (#21037)
+* examples/lorawan: drop crazy STM32 hack (#20883)
+* gcoap/forward_proxy: handle timeout case (#20915)
+* gh-actions: bump upload-artifact to v4 (#20875)
```
Comment (miri64) [inline release-notes.txt:218]: Why is this here but `gh-actions: bump deprecated packages (#20892)` is under `Testing`?

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,510 @@
...
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2024.10 release includes:
```
Comment (mguetschow) [inline release-notes.txt:21]:
```suggestion
Highlights of the 2024.10 release:
```

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,510 @@
...
+
+This release fixes some long standing pktbuf leaks that occurred when using
```
Comment (mguetschow) [inline release-notes.txt:23]:
```suggestion
This release fixes some long-standing packet buffer leaks that occurred when using
```

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,510 @@
...
+`gnrc_netif_pktq` and 6LoWPAN or `netdev_new_api` (#20983, #20834).
+
+With SUIT it is now easier to accept firmwares signed with different keys,
+e.g. a node build and signed with a development key can be configured to
```
Comment (mguetschow) [inline release-notes.txt:27]:
```suggestion
e.g., a node build and signed with a development key can be configured to
```

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,510 @@
...
+accept images signed with either the development key or the production key (#21019).
+The public keys are now also stored on the developer's machine, so if a
+encrypted private key is used, you don't need the decryption password to build
+a firmware, only for signing it.
```
Comment (mguetschow) [inline release-notes.txt:31]: Is there a PR number for this part, too?

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,510 @@
...
++ boards/nucleo-g474re: add MCU table (#20851)
++ boards/nucleo64: add pinout source (#20931)
++ boards/stm32l476g-disco: add ADC support (#20793)
++ boards/stm32l476g-disco: add PWM support (#20807)
++ examples: Add asynchronous Rust example (#20830)
+* boards/ek-lm4f120x: Change internal LED macro for C2Rust
+  compatibility (#20831)
+* boards/nucleo-l432kc: enable CAN support (#20845)
+* CI: fix true and false positives by newer codespell version (#20899)
+* drivers/led: Allow LEDn_ON to be disabled by other modules (#20833)
```
Comment (mguetschow) [inline release-notes.txt:112]: move to drivers?

File: release-notes.txt [hunk 10]
```diff
@@ -1,3 +1,510 @@
...
+
+CPU
+---
+
+* core/thread: always use THREAD_CREATE_STACKTEST when DEVELHELP is
+  enabled (#20450)
+* cpu/cc2538: mask length byte before checking CRC [backport 2024.10] (#21038)
```
Comment (mguetschow) [inline release-notes.txt:119]:
```suggestion
* cpu/cc2538: mask length byte before checking CRC (#21038)
```

and probably in other places, too

**Review by mguetschow — CHANGES_REQUESTED** [review]

**Review by Teufelchen1 — APPROVED** [review]
Nice!

**Review by mguetschow — APPROVED** [review]
Why FlexTape? :)

File: release-notes.txt [hunk 11]
```diff
@@ -1,3 +1,511 @@
...
+This release fixes some long-standing packet buffer leaks that occurred when using
+`gnrc_netif_pktq` and 6LoWPAN or `netdev_new_api` (#20983, #20834).
+
+A long standing-bug in the CC2538 radio driver has been fixed when processing 
+corrupted frames, that before could render the device unresponsive. This can be 
+triggered due to radio interference, but we also believe that an attacker within range 
+of the radio could send crafted frames to reliably trigger the issue. We therefore 
+consider this a denial of service vulnerability and advise all users of the CC2538 to 
+upgrade to 2024.10 or cherry-pick 1b61216da734fcd40ae0d832b800e4e67975bdc8. More 
+details are published in the [security advisory](https://github.com/RIOT-OS/RIOT/security/advisories/GHSA-m75q-8vj8-wppw).
```
Comment (maribu) [inline release-notes.txt:32]:
```suggestion
A long standing-bug in the CC2538 radio driver has been fixed when processing
corrupted frames, that before could render the device unresponsive. This can be
triggered due to radio interference, but we also believe that an attacker within range
of the radio could send crafted frames to reliably trigger the issue. We therefore
consider this a denial of service vulnerability and advise all users of the CC2538 to
upgrade to 2024.10 or cherry-pick 1b61216da734fcd40ae0d832b800e4e67975bdc8. More
details are published in the [security advisory](https://github.com/RIOT-OS/RIOT/security/advisories/GHSA-m75q-8vj8-wppw).
```

**Comment by benpicco** [comment]
> Why FlexTape? 

because of that nasty memory leak ;)

**Review by maribu — APPROVED** [review]
